nix/pkgs/main/default.nix

```diff
@@ -6,20 +6,73 @@
 , pkgsBuildHost
 , rocksdb
 , rust
+, rust-jemalloc-sys
 , stdenv
 
 # Options (keep sorted)
 , default_features ? true
+, all_features ? false
 , features ? []
 , profile ? "release"
 }:
 
 let
+# We perform default-feature unification in nix, because some of the dependencies
+# on the nix side depend on feature values.
+workspaceMembers = builtins.map (member: "${inputs.self}/src/${member}")
+  (builtins.attrNames (builtins.readDir "${inputs.self}/src"));
+crateFeatures = path:
+  let manifest = lib.importTOML "${path}/Cargo.toml"; in
+  lib.remove "default" (lib.attrNames manifest.features) ++
+  lib.attrNames
+    (lib.filterAttrs
+      (_: dependency: dependency.optional or false)
+      manifest.dependencies);
+crateDefaultFeatures = path:
+  (lib.importTOML "${path}/Cargo.toml").features.default;
+allDefaultFeatures = lib.unique
+  (lib.flatten (builtins.map crateDefaultFeatures workspaceMembers));
+allFeatures = lib.unique
+  (lib.flatten (builtins.map crateFeatures workspaceMembers));
+features' = lib.unique
+  (features ++
+    lib.optionals default_features allDefaultFeatures ++
+    lib.optionals all_features allFeatures);
+
+featureEnabled = feature : builtins.elem feature features';
+
+# This derivation will set the JEMALLOC_OVERRIDE variable, causing the
+# tikv-jemalloc-sys crate to use the nixpkgs jemalloc instead of building it's
+# own. In order for this to work, we need to set flags on the build that match
+# whatever flags tikv-jemalloc-sys was going to use. These are dependent on
+# which features we enable in tikv-jemalloc-sys.
+rust-jemalloc-sys' = (rust-jemalloc-sys.override {
+  # tikv-jemalloc-zest/unprefixed_malloc_on_supported_platforms feature
+  unprefixed = true;
+}).overrideAttrs (old: {
+  configureFlags = old.configureFlags ++
+    # tikv-jemalloc-sys/stats feature
+    [ "--enable-stats" ] ++
```

Comment:
Strawberry pointed out that `stats` is actually enabled by default, and we need to pass `--disable-stats` to disable. She made a PR to fix a similar issue in tikv jemallocator: https://github.com/tikv/jemallocator/pull/82.

Reply:
Turns out that when I wrote this code, I was looking at an older commit on `main`, where we were enabling the `tikv-jemalloc-sys/stats` feature unconditionally. The feature was actually removed entirely in 6c1434c1657c5a911c98a9cfbe06c984d4079690, which was likely unintentional.

I've removed the `--enable-stats` flag from this PR. We can add it back when we reenable the `tikv-jemalloc-sys/stats` feature.